sklearn/ensemble/majorityrule_classifier.py

```diff
@@ -0,0 +1,166 @@
+eclf = MajorityRuleClassifier(clfs=[clf1, clf2, clf3])
+eclf.fit(X_train, y_train)
+eclf.fit(X_test)
+
+## Example 2: Using Average Probabilties
+# initialization for using average predict_proba scores to derive the class label.
+eclf = MajorityRuleClassifier(clfs=[clf1, clf2, clf3], weights=[1,1,1])
+eclf.fit(X_train, y_train)
+eclf.fit(X_test)
+
+
+"""
+
+# Authors: Sebastian Raschka <se.raschka@gmail.com>
+#
+# Licence: BSD 3 clause
+
+from ..base import BaseEstimator
+from ..base import ClassifierMixin
+from ..base import TransformerMixin
+import numpy as np
+import operator
+
+class MajorityRuleClassifier(BaseEstimator, ClassifierMixin, TransformerMixin):
+    """ 
+    Majority rule classifier for scikit-learn estimators.
+        
+    Parameters
+    ----------
+    clfs : array-like, shape = [n_classifiers]
+      A list of scikit-learn classifier objects.
+      
+    weights : array-like, shape = [n_classifiers], optional (default=`None`)
+      If `None`, the majority rule voting will be applied to the predicted 
+      class labels. If a list of weights (`float` or `int`) is provided, 
+      the averaged raw probabilities (via `predict_proba`) will be used to 
+      determine the most confident class label.
+        
+    Attributes
+    ----------
+    classes_ : array-like, shape = [n_class_labels, n_classifiers]
+        Class labels predicted by each classifier if `weights=None`.
+       
+    probas_ : array, shape = [n_probabilities, n_classifiers]
+        Predicted probabilities by each classifier if `weights=array-like`.
+      
+    """
+    def __init__(self, clfs, weights=None):
+        self.clfs = clfs
+        self.weights = weights
+
+    def fit(self, X, y):
+        """ 
+        Fits the scikit-learn estimators.
+        
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+            
+        y : array-like, shape = [n_samples]
+            Target values.
+      
+        Returns
+        -------
+        self : object
+        
+        """
+        for clf in self.clfs:
+            clf.fit(X, y)
+            
+        return self
+            
+    def predict(self, X):
+        """
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        
+        Returns
+        ----------
+        maj : array-like, shape = [n_class_labels]
+            Predicted class labels by majority rule.
+        
+        """
+        if self.weights:
+            avg = self.predict_proba(X)
+            maj = np.apply_along_axis(lambda x: max(enumerate(x), key=operator.itemgetter(1))[0], axis=1, arr=avg)
```

Comment:
Could you refactor this to two lines maybe? 
( PEP8 specifications say that max line length should be 79... )